src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -219,11 +219,8 @@ public static object CreateEnumerableValue(ref ReadStack state)
             }
             else
             {
-                // Could not create an instance to be returned. For derived types, this means there is no parameterless ctor.
-                throw ThrowHelper.GetNotSupportedException_SerializationNotSupportedCollection(
-                    jsonPropertyInfo.DeclaredPropertyType,
-                    jsonPropertyInfo.ParentClassType,
-                    jsonPropertyInfo.PropertyInfo);
+                ThrowHelper.ThrowNotSupportedException_DeserializeCreateObjectDelegateIsNull(jsonPropertyInfo.DeclaredPropertyType);
+                return null;
```

Comment:
You wouldn't need the `return null` if you inverted the `if`, and it might actually read a little cleaner:
```C#
if (runtimeClassInfo.CreateObject == null)
{
    ThrowHelper.ThrowNotSupportedException_DeserializeCreateObjectDelegateIsNull(jsonPropertyInfo.DeclaredPropertyType);
}

return runtimeClassInfo.CreateObject();
```